services/config/src/useDate.ts

```diff
@@ -0,0 +1,134 @@
+import { useCallback, useContext, useMemo } from 'react'
+import { ConfigContext } from './ConfigContext'
+import { DateComponents, DateInput } from './types'
+
+// extend date with extra methods
+class DHIS2Date extends Date {
+    serverOffset: number
+    serverTimezone: string
+    clientTimezone: string
+
+    constructor({
+        date,
+        serverOffset,
+        serverTimezone,
+        clientTimezone,
+    }: {
+        date: DateInput
+        serverOffset: number
+        serverTimezone: string
+        clientTimezone: string
+    }) {
+        if (date) {
+            super(date)
+        } else {
```

Comment:
to avoid `super(undefined)` which will result in an Invalid Date being returned. This might be better as a one-liner `super(date ?? null)`, but I'm unsure if we want this at all

Reply:
I think the current implementation is perfectly fine, assuming that we want to initialise the date to the current date/time if no `date` is passed. `super(date ?? null)` will set the date to 1 January 1970, so we definitely don't want that.

An alternative approach could be to give the date field a default value of `Date.now`, and then you can call `super(date)` unconditionally. Mind you, then you'd also have to ensure the options object passed to the constructor also defaults to an empty object.